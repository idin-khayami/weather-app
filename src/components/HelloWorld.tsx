import React from 'react';

const HelloWorld: React.FC = () => {
  return <div>Hello Here</div>;
};

export default HelloWorld;
