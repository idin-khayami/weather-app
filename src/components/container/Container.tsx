import { ReactNode } from 'react';

import clsx from 'clsx';

import { ComponentBaseProps } from '@/types/global.types';

import styles from './container.module.scss';

interface ContainerProps extends ComponentBaseProps {
  children: ReactNode;
}

export const Container = ({
  dataTestId,
  children,
  className,
}: ContainerProps) => {
  return (
    <div
      data-testid={dataTestId}
      className={clsx([styles.container, className])}
    >
      {children}
    </div>
  );
};
