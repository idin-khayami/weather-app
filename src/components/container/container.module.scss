@import '../../styles/vars';

.container {
  margin: 0 auto;
  width: 100%;

  @include media-query-min($sm) {
    max-width: $sm;
  }

  @include media-query-min($md) {
    max-width: $md;
  }

  @include media-query-min($lg) {
    max-width: $lg;
  }

  @include media-query-min($xl) {
    max-width: $xl;
  }

  @include media-query-min($xxl) {
    max-width: $xxl;
  }
}
