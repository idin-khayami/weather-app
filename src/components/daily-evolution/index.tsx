import { DailyEvolution } from '@/components/daily-evolution/DailyEvolution';

export { DailyEvolution };
