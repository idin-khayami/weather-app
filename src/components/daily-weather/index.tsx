import { DailyWeather } from '@/components/daily-weather/DailyWeather';

export { DailyWeather };
