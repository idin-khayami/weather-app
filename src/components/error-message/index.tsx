import { ErrorMessage } from '@/components/error-message/ErrorMessage';

export { ErrorMessage };
