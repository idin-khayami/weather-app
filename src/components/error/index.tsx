import { Error } from '@/components/error/Error';

export { Error };
