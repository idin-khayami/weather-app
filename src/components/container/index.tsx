import { Container } from '@/components/container/Container';

export { Container };
