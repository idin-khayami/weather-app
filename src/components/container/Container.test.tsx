import { render, screen } from '@testing-library/react';

import { Container } from '@/components/container';

describe('Component: container', () => {
  it('should match the snapshot', () => {
    render(<Container dataTestId="testingContainer">Content</Container>);

    const cont = screen.getByTestId('testingContainer');

    expect(cont).toMatchSnapshot();
  });
});
