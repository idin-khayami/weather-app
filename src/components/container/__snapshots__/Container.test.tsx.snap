// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: container should match the snapshot 1`] = `
<div
  class="container"
  data-testid="testingContainer"
>
  Content
</div>
`;
